src/core/2d_primitives.js

```diff
@@ -13,6 +13,10 @@ var constants = require('./constants');
 var canvas = require('./canvas');
 require('./error_helpers');
 
+// Make mode adjust functions available for testing.
+p5.prototype._modeAdjust = canvas.modeAdjust;
+p5.prototype._arcModeAdjust = canvas.arcModeAdjust;
```

Comment:
Yeah it’s just for testing… I spent quite a long time trying to find a more elegant solution but ultimately gave up. If anyone can think of an alternative that would be great!